osu.Game/Rulesets/Mods/ModHidden.cs

```diff
@@ -38,7 +38,13 @@ public void ReadFromConfig(OsuConfigManager config)
         public virtual void ApplyToDrawableHitObjects(IEnumerable<DrawableHitObject> drawables)
         {
             if (IncreaseFirstObjectVisibility.Value)
+            {
+                var firstObject = drawables.FirstOrDefault();
```

Comment:
I believe this should be

```suggestion
                var firstObject = drawables.FirstOrDefault(IsFirstHideableObject);
```

or, probably better, rewrite this logic a bit to be

```csharp
if (IncreaseFirstObjectVisibility.Value)
{
    drawables = drawables.SkipWhile(h => !IsFirstHideableObject(h));

    var firstObject = drawables.FirstOrDefault();
    if (firstObject != null)
        firstObject.ApplyCustomUpdateState += ApplyFirstObjectIncreaseVisibilityState;

    drawables = drawables.Skip(1);
}
```

Can be tested on any map that starts with a spinner - the second object won't fade out at all while the third will work like normal hidden.